packages/provider/pkg/provider/schema.go

```
@@ -293,23 +293,10 @@ func putSchemaIfChanged(ctx *pulumi.Context, policyStoreId string, cedarJSON str
     }
     client := vpapi.NewFromConfig(cfg)
 
-    // Fetch current schema; treat NotFound as empty
-    var current string
+    // Use internal wrapper for idempotent PutSchema.
+    // Construct client and delegate. This centralizes JSON minification and comparisons.
     pulumiCtx := ctx.Context()
-    getOut, err := client.GetSchema(pulumiCtx, &vpapi.GetSchemaInput{PolicyStoreId: &policyStoreId})
-    if err == nil && getOut.Schema != nil {
-        current = *getOut.Schema
-    }
-    if normalizeJson(current) == normalizeJson(cedarJSON) {
-        ctx.Log.Info("AVP: schema unchanged; skipping PutSchema", &pulumi.LogArgs{})
-        return nil
-    }
-    // Apply
-    _, err = client.PutSchema(pulumiCtx, &vpapi.PutSchemaInput{
-        PolicyStoreId: &policyStoreId,
-        Definition:    &vpapiTypes.SchemaDefinitionMemberCedarJson{Value: cedarJSON},
-    })
-    if err != nil {
+    if err := vpwrap.PutSchemaIfChanged(pulumiCtx, client, policyStoreId, cedarJSON); err != nil {
         return fmt.Errorf("failed to put schema: %w", err)
```

Does this throw?  We do not want to swallow errors with logging only do we?  I would like to adopt idiomatic error handling logic for Go with a preference to bubbling up errors to be handled only by code that has appropriate context to handle the error